client/src/js/modules/dc/views/gridplot.js

```diff
@@ -514,7 +524,16 @@ define(['jquery', 'marionette',
 
         _getVal: function(pos) {
             var val = null
-            const d = Number(this.ui.ty.val()) > -1 ?  this.distl.get('data')[Number(this.ui.ty.val())] : []
+            var d = []
+            if (this.ui.ty.is(":visible")) {
+                d = Number(this.ui.ty.val()) > -1 ? this.distl.get('data')[Number(this.ui.ty.val())] : []
+            }
+            if (this.ui.ty2.is(":visible")) {
+                var a = this.attachments.findWhere({ 'DATACOLLECTIONFILEATTACHMENTID': this.ui.ty2.val() })
```

Comment:
```suggestion
                let a = this.attachments.findWhere({ 'DATACOLLECTIONFILEATTACHMENTID': this.ui.ty2.val() })
```
Also, would there be a more descriptive name for this variable?